### PR TITLE
Enable Langchain on Intel GPU

### PR DESCRIPTION
**Pull request by ivy-lv11**
Enable Langchain on Intel GPU

### CODE REVIEW

File: libs/community/tests/integration_tests/llms/test_ipex_llm.py [hunk 1]
```diff
@@ -86,3 +86,23 @@ def test_save_load_lowbit(model_id: str) -> None:
...
+        model_kwargs={
+            "temperature": 0,
+            "max_length": 16,
+            "trust_remote_code": True,
+        },
+        device_map="xpu",
+    )
+    output = llm.generate(["Hello!"])
+    assert isinstance(output, LLMResult)
+    assert isinstance(output.generations, list)
```
Comment (Oscilloscope98) [inline libs/community/tests/integration_tests/llms/test_ipex_llm.py]: For embedding, I also treat device as an env variable: https://github.com/langchain-ai/langchain/blob/6bb9138b84bffe06e97e007082b0034e9666d4de/libs/community/tests/integration_tests/embeddings/test_ipex_llm.py#L16

Not sure whether we should unify here :)

File: libs/community/langchain_community/llms/ipex_llm.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import logging
```
Comment (Oscilloscope98) [inline libs/community/langchain_community/llms/ipex_llm.py]: Do not forget to fix the style errors :)

Comment (ivy-lv11) [reply libs/community/langchain_community/llms/ipex_llm.py]: Fixed.

**Comment by shane-huang** [comment]
there're still format errors. Need to fix them until they're all passed.

**Comment by shane-huang** [comment]
Change the title of the title of the original notebook to `IPEX-LLM on Intel CPU` https://github.com/intel-analytics/langchain/blob/ipex-llm-llm-gpu/docs/docs/integrations/llms/ipex_llm.ipynb. And also modify the 1st sentence

> This example goes over how to use LangChain to interact with ipex-llm for text generation.
to 
> This example goes over how to use LangChain to interact with ipex-llm for text generation on Intel CPU.

File: docs/docs/integrations/llms/ipex_llm_gpu.ipynb [hunk 3]
```diff
@@ -0,0 +1,250 @@
+{
```
Comment (Oscilloscope98) [inline docs/docs/integrations/llms/ipex_llm_gpu.ipynb:1]: Seems like you forgot to change the usage in this doc :) regarding device

File: docs/docs/integrations/llms/ipex_llm_gpu.ipynb [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+    "question = \"What is AI?\"\n",
+    "output = llm_chain.invoke(question)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Save/Load Low-bit Model\n",
+    "Alternatively, you might save the low-bit model to disk once and use `from_model_id_low_bit` instead of `from_model_id` to reload it for later use - even across different machines. It is space-efficient, as the low-bit model demands significantly less disk space than the original model. And `from_model_id_low_bit` is also more efficient than `from_model_id` in terms of speed and memory usage, as it skips the model conversion step. You can similarly set `device` to `xpu` in `model_kwargs` in order to load the LLM model to Intel GPU. "
```
Comment (Oscilloscope98) [inline docs/docs/integrations/llms/ipex_llm_gpu.ipynb]: `xpu` - > `"xpu"`

**Review by Oscilloscope98 — APPROVED** [review]
LGTM

File: libs/community/tests/integration_tests/llms/test_ipex_llm.py [hunk 5]
```diff
@@ -12,12 +13,18 @@
     not model_ids_to_test, reason="TEST_IPEXLLM_MODEL_IDS environment variable not set."
 )
 model_ids_to_test = [model_id.strip() for model_id in model_ids_to_test.split(",")]  # type: ignore
+device = os.getenv("TEST_IPEXLLM_BGE_EMBEDDING_MODEL_DEVICE") or "cpu"
```
Comment (Oscilloscope98) [inline libs/community/tests/integration_tests/llms/test_ipex_llm.py]: TEST_IPEXLLM_MODEL_DEVICE

**Pull request merged**